content/en/contribute/code/android/development-setup.md

```diff
@@ -162,6 +162,20 @@ _Only CHT Android_
 
 Refer to the [CHT Core Developer Guide]({{< relref "contribute/code/core/dev-environment#nginx-local-ip" >}}).
 
+#### Manually testing with older Android versions
+
+_Only CHT Android_
+
+Later versions of the CHT, only support [running on Chrome/Webview 90+]({{< relref "core/releases#dependencies" >}}). Some older versions of Android will not have a new enough Android System Webview to be able to run the CHT webapp. This is particularly true of emulated Android devices, which typically are configured to not receive updates. To test the CHT functionality on these devices you must manually upgrade the version of the Chrome/Webview apk used as the Android System Webview.
+
+To upgrade the necessary apk on an emulated Android device:
+
+- Start by using the `Google APIs` variant of the Android image. The base AOSP image bakes in the `com.android.webview` package, making it difficult to update.
+- Depending on your version of Android, you will need to manually download a new version of the apk (make sure to get the correct variant for your architecture):
+    - For Android 7-9, you should download a new version of `com.android.chrome` (e.g. from [apkMirror](https://www.apkmirror.com/apk/google-inc/chrome/)).
+    - For Android 10+, you should download a new version of `com.google.android.webview` (e.g. from [apkMirror](https://www.apkmirror.com/apk/google-inc/android-system-webview/)).
```

Comment:
What's the steps for Android 5-6? I've tried installing APKs of both Chrome and Android System WebView, both versions 90+, and the CHT app would freeze on the loading screen with the following error being logged by the browser:
```
https://gamma.dev.medicmobile.org/scripts.js:1 | Uncaught SyntaxError: Unexpected token {
```
The emulated device is running Android 5.0 with Google Play Services, I installed the APKs from APKMirror using `adb` like shown below

The funny thing is that opening the same instance directly in Chrome worked...

Reply:
Yep. After hours of tinkering I have not come up with any solution for Android 5-6.  For these Android versions, it seems like even the "Google API's" images have the `com.android.webview` package baked in. Technically, it should be possible to upgrade that package, but I guess it might require actually editing the system APKs (e.g. the device carrier releasing an updated ROM with a newer WebView).  It seems of dubious value to invest much more effort into getting this to work since I am a bit skeptical of finding a device in the wild that is Android 5 with WebView 90+....

_However_, looking at the [support matrix](https://docs.communityhealthtoolkit.org/core/releases/), it does seem like a more likely use-case we should perhaps test for is connecting the latest cht-android on an Android 5 device to a CHT `3.x` instance. Theoretically we would just need WebView 53+ for this (though it is worth noting that the emulated Android 5 device seems to still only have WebView 39....). 

I am going to try and do some more testing and maybe see about adding some more docs here....

Reply:
I gave it a shot on my HTC One S running Android 5 (rooted cyanogenmod) & WebView 95 and it's working normally with a CHT 4.5 instance. Not sure why the HTC picks up on the WebView APK I installed but not the emulator though...

Reply:
> cyanogenmod

:star_struck: Ahh the good ol days!

Reply:
Okay, I have added a bit more info about testing with older Android versions.  Let me know if you think there is anything else we can clarify here!